MREUnityRuntime/MREUnityRuntimeLib/Core/PhysicsBridge.cs

```diff
@@ -0,0 +1,310 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using MixedRealityExtension.Core.Physics;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace MixedRealityExtension.Core
+{
+	public class RigidBodyPhysicsBridgeInfo
+	{
+		public RigidBodyPhysicsBridgeInfo(Guid id, UnityEngine.Rigidbody rb, bool ownership)
+		{
+			Id = id;
+			RigidBody = rb;
+			Ownership = ownership;
+			lastTimeKeyFramedUpdate = 0.0f;
+			lastValidLinerVelocity.Set(0.0f, 0.0f, 0.0f);
+			lastValidAngularVelocity.Set(0.0f, 0.0f, 0.0f);
+
+			IsKeyframed = false;
+		}
+
+		public Guid Id;
+
+		public UnityEngine.Rigidbody RigidBody;
+
+		/// these 3 fields are used to store the actual velocities
+		public float lastTimeKeyFramedUpdate;
+		public UnityEngine.Vector3 lastValidLinerVelocity;
+		public UnityEngine.Vector3 lastValidAngularVelocity;
+
+		/// true if this rigid body is owned by this client
+		public bool Ownership;
+
+		public bool IsKeyframed;
+	}
+
+	/// <summary>
+	/// the main class that is the bridge between the MRE Unity and the networked physics logic
+	/// </summary>
+	public class PhysicsBridge
+	{
+
+		private int _countOwnedTransforms = 0;
+		private int _countStreamedTransforms = 0;
+
+		private SortedList<Guid, RigidBodyPhysicsBridgeInfo> _rigidBodies = new SortedList<Guid, RigidBodyPhysicsBridgeInfo>();
+
+		TimeSnapshotManager _snapshotManager = new TimeSnapshotManager();
+
+		/// the prediction object
+		IPrediction _predictor = new PredictionInterpolation();
+
+		public PhysicsBridge()
+		{
+		}
+
+
+		#region Angle Helper Function
+		/// helper function to transform an Euler angle to radian
+		public static float TransformEulerAngleToRadian(float eulerAngle)
+		{
+			float ret = eulerAngle;
+			ret = (ret >= 180) ? (ret - 360.0f) : ret;
+			ret = (ret <= -180) ? (360 + ret) : ret;
+			ret = ret * ((float)Math.PI / 180.0f);
+			return ret;
+		}
+		/// helper function to transform an Euler angle to radians
+		public static UnityEngine.Vector3 TransformEulerAnglesToRadians(UnityEngine.Vector3 eulerAngles)
+		{
+			UnityEngine.Vector3 ret = eulerAngles;
+			ret.x = TransformEulerAngleToRadian(ret.x);
+			ret.y = TransformEulerAngleToRadian(ret.y);
+			ret.z = TransformEulerAngleToRadian(ret.z);
+			return ret;
+		}
```

Comment:
These are helpful utility methods and should go with the rest of our utility methods in the UtilMethods.cs file.